### PR TITLE
Annotate `__iter__` method of Series

### LINKED ISSUE #119: __iter__ method of a Series should be type annotated
**Describe the bug**

A type checker can not correctly resolve the return type of the `__iter__` method of a pandas series.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```python
from typing_extensions import reveal_type
import pandas as pd

series = pd.Series([1, 2, 3], dtype=int)
reveal_type(series) # type = Series[int]
iterable = iter(series)
reveal_type(iterable) # type = Unkown, type should be Iterable[int]
```
2. Indicate which type checker you are using (`mypy` or  `pyright`).

I am using pyright 1.1.257

3. Pyright message
```
  /home/pandas_types.py:5:13 - information: Type of "series" is "Series[int]"
  /home/pandas_types.py:7:13 - information: Type of "iterable" is "Unknown"
```

**Please complete the following information:**
 - OS: Linux
 - OS Version: Ubuntu 22.04
 - python version: 3.10.4
 - version of type checker 1.1.257
 - version of installed `pandas-stubs` 1.4.3.220704

### PR DESCRIPTION
**Pull request by diepala**
- [x] Closes #119 (Replace xxxx with the Github issue number)
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: pandas-stubs/core/generic.pyi [hunk 1]
```diff
@@ -94,7 +94,6 @@ class NDFrame(PandasObject, indexing.IndexingMixin):
     def bool(self) -> _bool: ...
     def __abs__(self) -> NDFrame: ...
     def __round__(self, decimals: int = ...) -> NDFrame: ...
-    def __iter__(self) -> Iterator: ...
```
Comment (twoertwein) [inline pandas-stubs/core/generic.pyi:97]: I wouldn't mind keeping this: there might be people inheriting from `NDFrame`.

Comment (diepala) [reply pandas-stubs/core/generic.pyi:97]: @Dr-Irv suggested removing this. Check [this comment](https://github.com/pandas-dev/pandas-stubs/issues/119#issuecomment-1179557772).

Comment (Dr-Irv) [reply pandas-stubs/core/generic.pyi:97]: > I wouldn't mind keeping this: there might be people inheriting from `NDFrame`.

`NDFrame` isn't in the public API, so people should _not_ be doing that!

**Review by Dr-Irv — APPROVED** [review]
Thanks @diepala

**Pull request merged**